### PR TITLE
[SDEM-PFEM] Updating DEM-PFEM coupling to new Swimming DEM standard

### PR DESCRIPTION
**Pull request by ipouplana**

### CODE REVIEW

File: applications/SwimmingDEMApplication/python_scripts/swimming_DEM_analysis.py [hunk 1]
```diff
@@ -194,7 +194,7 @@ def SetBetaParameters(self):
         Add("store_fluid_pressure_option").SetBool(False)
         Add("laplacian_calculation_type").SetInt(0)
         Add("faxen_terms_type").SetInt(0)
-        Add("material_acceleration_calculation_type").SetInt(1)
+        Add("material_acceleration_calculation_type").SetInt(8)
```
Comment (GuillermoCasas) [inline applications/SwimmingDEMApplication/python_scripts/swimming_DEM_analysis.py]: Modify your own analysis

File: applications/SwimmingDEMApplication/tests/NightTests.py [hunk 2]
```diff
@@ -41,7 +46,8 @@ def SetTestSuite(suites):
           candelier_no_history_with_lift_test,
           candelier_no_history_non_inertial_test,
           candelier_with_history_test,
-          candelier_with_history_hinsberg_test
+          candelier_with_history_hinsberg_test,
+          Test
```
Comment (GuillermoCasas) [inline applications/SwimmingDEMApplication/tests/NightTests.py]: Use similar nomenclature

File: applications/SwimmingDEMApplication/python_scripts/DEM_coupled_pfem_fluid_dynamics_analysis.py [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+
+import KratosMultiphysics as Kratos
+import KratosMultiphysics.PfemFluidDynamicsApplication
+
+from pfem_fluid_dynamics_analysis import PfemFluidDynamicsAnalysis
+
+class DEMCoupledPFEMFluidDynamicsAnalysis(PfemFluidDynamicsAnalysis):
+
+    def __init__(self, model, parameters=None, pp=None):
```
Comment (GuillermoCasas) [inline applications/SwimmingDEMApplication/python_scripts/DEM_coupled_pfem_fluid_dynamics_analysis.py]: Here you should be passing the whole project_parameters and only pass to the parent the project_parameters["fluid_parameters"] bit. See [https://github.com/KratosMultiphysics/Kratos/blob/SDEM/debugging-and-cleaning-analysis/applications/SwimmingDEMApplication/python_scripts/DEM_coupled_fluid_dynamics_analysis.py](url).

**Review by GuillermoCasas — CHANGES_REQUESTED** [review]
See comments

**Review by jgonzalezusua — DISMISSED** [review]

File: applications/PfemFluidDynamicsApplication/python_scripts/pfem_fluid_dynamics_analysis.py [hunk 4]
```diff
@@ -0,0 +1,393 @@
...
+import KratosMultiphysics
+import KratosMultiphysics.ExternalSolversApplication
+import KratosMultiphysics.DelaunayMeshingApplication
+import KratosMultiphysics.PfemFluidDynamicsApplication
+import KratosMultiphysics.SolidMechanicsApplication
+
+import DEM_coupled_pfem_fluid_dynamics_analysis as fluid_solution
+import pfem_fluid_solver_analysis as pfem_solver
+
+from analysis_stage import AnalysisStage
```
Comment (philbucher) [inline applications/PfemFluidDynamicsApplication/python_scripts/pfem_fluid_dynamics_analysis.py]: it is more future-proof to do the following:
```suggestion
from KratosMultiphysics.analysis_stage import AnalysisStage
```

**Review by GuillermoCasas — DISMISSED** [review]
I am approving, but please wait for my next pull request to apply more changes to the application

**Review by jgonzalezusua — DISMISSED** [review]

**Review by jgonzalezusua — DISMISSED** [review]

File: applications/SwimmingDEMApplication/python_scripts/swimming_DEM_PFEM_analysis.py [hunk 5]
```diff
@@ -5,36 +5,28 @@
...
-        #parameters_file = open("ProjectParametersDEM.json",'r')
-        #self.project_parameters = Parameters(parameters_file.read())
-        #self.SetDoSolveDEMVariable()
-        #self.pp.Dt = self.fluid_solution.GetDeltaTimeFromParameters()
-        #self.SetBetaParameters()
-        #self.SetCustomBetaParameters(varying_parameters)
-        #self.pp.domain_size = self.fluid_solution.ProjectParameters["problem_data"]["domain_size"].GetInt()
-        #super(Algorithm,self).SetCouplingParameters(varying_parameters)
+        self.project_parameters["material_acceleration_calculation_type"].SetInt(8)
+        #self.vars_man.Dt = self.project_parameters['fluid_parameters']["problem_data"]["time_step"]
```
Comment (GuillermoCasas) [inline applications/SwimmingDEMApplication/python_scripts/swimming_DEM_PFEM_analysis.py]: Please remove comments

File: applications/SwimmingDEMApplication/python_scripts/swimming_DEM_procedures.py [hunk 6]
```diff
@@ -78,6 +78,7 @@ def InitializeVariablesWithNonZeroValues(parameters, fluid_model_part, balls_mod
         SetModelPartSolutionStepValue(fluid_model_part, FLUID_FRACTION_OLD, 1.0)
     if checker.ModelPartHasNodalVariableOrNot(balls_model_part, FLUID_FRACTION_PROJECTED):
         SetModelPartSolutionStepValue(balls_model_part, FLUID_FRACTION_PROJECTED, 1.0)
+        SetModelPartSolutionStepValue(balls_model_part, BASSET_FORCE, Vector([0.]*3))
```
Comment (GuillermoCasas) [inline applications/SwimmingDEMApplication/python_scripts/swimming_DEM_procedures.py]: This should not be needed... Why is it not initially 0? Could you investigate this?

File: applications/SwimmingDEMApplication/python_scripts/variables_management.py [hunk 7]
```diff
@@ -199,7 +199,7 @@ def ConstructListsOfVariables(self, parameters):
         if parameters["virtual_mass_force_type"].GetInt() > 0 and parameters["add_each_hydro_force_option"].GetBool():
             self.dem_vars += [VIRTUAL_MASS_FORCE]
 
-        if parameters["basset_force_type"].GetInt() > 0 and parameters["add_each_hydro_force_option"].GetBool():
+        if True or parameters["basset_force_type"].GetInt() > 0 and parameters["add_each_hydro_force_option"].GetBool():
```
Comment (GuillermoCasas) [inline applications/SwimmingDEMApplication/python_scripts/variables_management.py]: For the moment just remove the 'if' and add it always please

File: applications/SwimmingDEMApplication/python_scripts/variables_management.py [hunk 8]
```diff
@@ -332,6 +332,7 @@ def ConstructListsOfResultsToPrint(self, parameters):
         self.variables_to_print_in_file = ["DRAG_FORCE", "LIFT_FORCE", "BUOYANCY", "VELOCITY"]
 
         self.dem_printing_vars = []
+
```
Comment (GuillermoCasas) [inline applications/SwimmingDEMApplication/python_scripts/variables_management.py:342]: remove space

**Review by GuillermoCasas — APPROVED** [review]

**Pull request merged**